nrf_security/src/drivers/builtin/Kconfig

```diff
@@ -237,7 +237,8 @@ config MBEDTLS_PSA_BUILTIN_KEY_TYPE_ECC_KEY_PAIR
 	bool
 	#prompt "PSA Key type ECC key pair support - builtin" if !PSA_PROMPTLESS
 	depends on PSA_WANT_KEY_TYPE_ECC_KEY_PAIR
-	default y if !PSA_DEFAULT_OFF
+	default y if !PSA_DEFAULT_OFF && \
```

Comment:
Nothing wrong as such on this line.

Just happen to notice location of this Kconfig file:
`nrf_security/src/drivers/builtin/Kconfig`

Since when have we started to have Kconfig files in source code directories ?
@SebastianBoe have you noticed this before ?

Reply:
I think this follows prior-art.

Principle to follow is that Kconfig should be as close as possible to what it's configuring.

So having Kconfig next to the C source file that it is configuring is OK.

Reply:
yeah, but usually (upstream at least) we just have folders with files, like this:
https://github.com/zephyrproject-rtos/zephyr/tree/main/subsys/net/l2/ethernet
it doesn't split into a dedicated source folder.

but nRF security changes that by splitting into include and src folders:
https://github.com/nrfconnect/sdk-nrfxlib/tree/main/nrf_security/src
https://github.com/nrfconnect/sdk-nrfxlib/tree/main/nrf_security/include

so when doing so, does Kconfig really belong in `src` ?

Not to mention the fact the we suddenly end-up having include folders down the source hierarchy, like here:
https://github.com/nrfconnect/sdk-nrfxlib/tree/main/nrf_security/src/drivers/nrf_cc3xx/include

I think splitting into `include` and `src` can make a lot of sense, but the way it's being done now in nRF Security is messy, as src is not just source files, but apparently also `includes` plus a lot more.

This is not a blocker for this PR, I just happened to notice this now.
And I feel this starts to look very messy, and not well structured.

Reply:
I think it's the choice of the dir name "src" that is confusing. In hindsight, it should have been named differently
as it communicates things that are not true for this src directory.

Reply:
completely agree with that, and that was the point I was trying to make initially.